docs/architecture/adr-059-evidence-composition-and-lifecycle.md

```diff
@@ -0,0 +1,274 @@
+
+- Fetch the trusted signed header at the same height as the conflicting header and compare with the conflicting header to work out which type of attack it is and in doing so return the malicious validators.
+
+  - If equivocation the validators that signed for the commits of both the trusted and signed header
+
+  - If lunatic the validators from the common val set that signed in the conflicting block
+
+  - If amnesia none (we can't know which validators are malicious). This also means that we don't
+  send amnesia evidence to the application
+
+- For each validator, check the look up table to make sure there already isn't evidence against this validator
+
+After verification we persist the evidence with the following key: `height/hash` to the pending evidence database in the evidence pool with the following format
+
+```go
+type EvidenceInfo struct {
+  ev Evidence
+  time time.Time
+  validators []Validator
+  totalVotingPower int64
+}
+```
+
+`time`, `validators` and `totalVotingPower` are need to form the ABCI Evidence that we will come to later.
+
+
+#### Broadcasting and receiving evidence
+
+The evidence pool also runs a reactor that broadcasts the newly validated
+evidence to all connected peers.
+
+Receiving evidence from other evidence reactors works in the same manner as receiving evidence from the consensus engine or a light client.
+
+
+#### Proposing evidence on the block
+
+When it comes to prevoting and precomitting a proposal that contains evidence, the full node will once again
+call upon the evidence pool to verify the evidence using `CheckEvidence(ev []Evidence)`:
+
+This performs the following actions:
+
+1. Loops through all the evidence to check that nothing has been duplicated
+
+2. For each evidence, run `fastCheck(ev evidence)` which works similar to `Has` but instead for `LightClientAttackEvidence` if it has the
+same hash it then goes on to check that the validators it has are all signers in the commit of the conflicting header. If it doesn't pass fast check (because it hasn't seen the evidence before) then it will have to verify the evidence.
+
+3. runs `Verify(ev Evidence)` - Note: this also saves the evidence to the db as mentioned before.
+
+
+#### Updating application and pool
+
+The final part of the lifecycle is when the block is committed and the `BlockExecutor` tries to apply state. Instead of trying to form the information necessary to produce the ABCI evidence itself, it gets the evidence pool to create it when it calls `Update(Block, State) []abci.Evidence`
+
+Here is the excerpt in the `ApplyBlock()` where the evidence pool is updated.
+```go
+abciResponses.BeginBlock.ByzantineValidators = evpool.Update(block, state)
+```
+
+The `Update()` function does the following:
+
+- Increments state which keeps track of both the current time and height used for measuring expiry
+
+- Marks evidence as committed and saves to db. This prevents validators from proposing committed evidence in the future
```

Comment:
```suggestion
- Marks evidence as committed and saves to db. This prevents validators from proposing committed evidence in the future.
```

As a heads up - markdown doesn't render newlines unless there are two of them back to back, or if you explicitly force them with a double space at the end of the line.

Reply:
I actually separated them so that it was easier to read in their unrendered form. You made an earlier comment about it and I thought it may referring to that. I also skip lines in other places that I used bullet points